### ISSUE TITLE
1 pixel bug: There is 1 white pixel in the center of my desktop background.

### ISSUE THREAD

**Issue by preeset** [linuxmint/sticky#64]
- Linux Mint 20.3, Notes version 1.8.
- Looked at code: /usr/lib/sticky/sticky.py
- Code contains [commit](https://github.com/linuxmint/sticky/commit/a3f97680db369a6894b63f74acaeef552b25c091)
- I have a white pixel in the center of my screen. Closing the application does not fix this.

How do I remove the pixel?

**Comment by collinss** [comment]
What desktop environment/window manager are you using? How are you closing the app?

**Comment by collinss** [comment]
Also, could you make sure the app is in fact not running after closing it? (ie through a system monitor, etc)

**Comment by Iss-in** [comment]
same, I'm using sway/Arch, on opening sticky app, beside the main window, one blank window is also displayed which can be closed but shouldn't show up

![image](https://user-images.githubusercontent.com/36916613/159171114-508242b0-b90c-4dba-81b1-0c4fbd3b4d5a.png)

In my case, size of this window is 75x60 pixels with title=Notes and app_id=sticky.py

**Comment by 63BeetleSmurf** [comment]
I've defiantly seen the single pixel showing before, but can't remember when. I was using Sticky on my work machine (Debian 11 with XFCE) but we moved to GNOME, so may have been messing around seeing what worked with that.

Anyway, I suspect it's just the dummy window showing up. Is it necessary to call show() for the dummy window? I took out that line and everything seems to work ok.

**Comment by Iss-in** [comment]
yes, commenting out this [line](https://github.com/linuxmint/sticky/blob/a2dff0d469611c4d4a42c053c4d4927816ca87b6/usr/lib/sticky/sticky.py#L676) removes the 1px dummy window. Had to make this change everytime I update the app, its somewhat annoying

**Comment by collinss** [comment]
@Iss-in I can't just get rid of it without consequences. It is used to be able to show a single entry in window lists and alt-tab (when the appropriate settings are enabled). There are some users who want to be able to access the notes using those methods, and the dummy window is the only way I've been able to accomplish that. I'm also not sure why it's still showing up in some desktop environments either. My suspicion is that those environments are ignoring certain requests to make the window invisible (such as moving it off screen, asking the window manager not to decorate it, and even overriding the theming to make everything invisible). I'm not sure if there is anything more I can do. It might need to be fixed at the window manager level, as those environments seem absolutely determined to show the user something (even if it's not intended in this case).

From the issue you described earlier, it sounds like sway is ignoring the request to not decorate the window (meaning it's putting borders, shadows, etc on the window even though it is specifically asked not to). If this is the case, then the problem is really with sway, and not the sticky note app. If a window requests not to be decorated, that request should be honored by the window manager, and any failure to do so is a bug (and there's nothing else I could do on my end anyway).

All that being said, I can't be sure that's what's happening, and unfortunately I don't have the time and resources to test and fix this for every desktop environment. If someone else wants to investigate further and finds a way to fix the issue on the sticky notes side, without causing regressions to current functionality, I'd be happy to merge it, but I don't have the time to go chasing down a bug in a desktop environment I don't have installed, and that has a high probability of not being fixable on my end.
